### PR TITLE
add components using in app

### PR DESCRIPTION
**Pull request by shivamSspirit**
add components using in app

### CODE REVIEW

File: src/components/auth/signup/signup.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+                        <h3 className="head-form ecom-user">User Signup</h3>
+                        <form id="my-form" onSubmit={handleSignSubmit}>
+                            <div className="med">
+                                <label className="label ecom-label" for="name"> Name </label>
+                                <input value={user?.name} onChange={handleUserSignup} required className="input" type="text" name="username" />
+                            </div>
+                            <div className="med">
+                                <label className="label ecom-label" for="Email"> Email </label>
+                                <input value={user?.email} onChange={handleUserSignup} required className="input input-email" type="email" name="email" />
+                                {/* <img className="form-icon form-icon-disable ecom-email" src="../../../images/svgs/email.svg"
```
Comment (aurobindoGupta) [inline src/components/auth/signup/signup.js:53]: remove

File: src/components/productlisting/productlisting.js [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+import { useGlobal } from '../../contexts/globalContext'
+
+import * as WishAPis from '../../api/wishlist'
+import * as CartApis from '../../api/cart'
+
+import * as ActionTypes from '../../constants/actions'
+
+// use proxy object object for validation
+// const person = {
+//     name: "John Doe",
```
Comment (aurobindoGupta) [inline src/components/productlisting/productlisting.js:19]: remove

**Review by aurobindoGupta — APPROVED** [review]
nice work.

File: src/components/auth/login/login.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
+import React, { useState, useContext, useEffect } from 'react'
+import './login.css'
+import * as AuthApis from '../../../api/authApi'
+import { Link, useNavigate } from 'react-router-dom'
+import { useAuth } from '../../../contexts/globalContext'
+import { useGlobal } from '../../../contexts/globalContext'
```
Comment (viru2001) [inline src/components/auth/login/login.js:6]: you can use [Absolute Imports](https://dev.to/mr_frontend/absolute-imports-in-create-react-app-3ge8). Imports will look clean

File: src/components/auth/signup/signup.css [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+}
+
+.redirect-signup{
+    margin-left: 2.8rem;
+}
+
+.ecom-login{
+    text-decoration: none;
+	color: var(--default-color);
+}
```
Comment (viru2001) [inline src/components/auth/signup/signup.css:19]: you can use prettier to properly format the indentation

File: src/components/cart/cart.js [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+        if (cartState.cartproducts) {
+            console.log('cartstate', cartState.cartproducts)
+        }
+    }, [cartState.cartproducts])
+
+    return (
+        <div>
+            <div className="cart-container">
+                <div>
+                    <h2 className="my-cart">My cart(3)</h2>
```
Comment (viru2001) [inline src/components/cart/cart.js:20]: is this hard coded ?

File: src/components/cart/cart.js [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+
+                    <div className="cart-partition">
+
+                        <div className="cart-products">
+                            <div className="align-card">
+                                {cartState.cartproducts && cartState.cartproducts?.map((item, idx) => (
```
Comment (viru2001) [inline src/components/cart/cart.js:26]: you can destructure `item` here

**Review by viru2001 — APPROVED** [review]

File: src/components/auth/login/login.js [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+        <>
+            <div class="login-container">
+                <div className="forms">
+                    <div className="single-input ecom-form">
+                        <h3 className="head-form ecom-user">User Login</h3>
+                        <form id="my-form" onSubmit={handleLogSubmit}>
+                            <div className="med">
+                                <label className="label ecom-label" for="Email"> Email </label>
+                                <input required value={user?.email} onChange={handleUserLogin} className="input input-email" type="email" name="email" />
+                                {/* <img className="form-icon form-icon-disable ecom-email" src="../../../images/svgs/email.svg"
```
Comment (MarkVed17) [inline src/components/auth/login/login.js:60]: Remove unnecessary code

**Pull request merged**